vue-js/src/components/task1/ListItem.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div id="list-item">
+    <div v-if="edit">
+      <li>{{item.text}}
+      <button v-on:click="setSavingMode">Edit</button>
+      <button  @click="$emit('delete-item')">Delete</button>
+      </li>
+    </div>
+    <div v-else>
+      <li>
+      <input id="new-item-text" v-model="item.text" />
+      <button v-on:click="saveNewText">Save</button>
+      <button v-on:click="cancelSavingMode">Cancel</button>
+      </li>
+    </div>
+    
+    
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'list-item',
+  props: ['item'],
+  data: function() {
+    return {
+      edit: true
+    }
+  },
+  methods: {
+    setSavingMode: function() {
+      this.edit = false;
+    },
+
+    cancelSavingMode: function() {
+      this.edit = true;
+    },
+    saveNewText: function() {
+      this.item.text = document.getElementById("new-item-text").value;

```

Comment:
If you would do it the Vue way™ you would create a new component that has the responsibility of adding a new todo. It would wrap the `input` and the submit `button` and you could just emit the whole data object.

Your solution with `document.getElementById` etc. relies on unique ids and classes and that's a common cause of bugs.